### PR TITLE
NAS-123274 / 23.10 / Remove usages of CoreService for getting disk temperatures

### PR DESCRIPTION
**Pull request by bvasilenko**
**Testing**

Emulating "Enclosure":
1. run `yarn ui mock -e`
2. run `yarn ui mock-enclosure -e`
3. navigate to **System Settings > Enclosure**

**Expected result:** Disk temperatures should still be fetched every N seconds, according to the new approach. So please test the closure page to make sure that temperatures are still updated properly. `

### CODE REVIEW

File: src/app/pages/system/view-enclosure/components/enclosure-disks/enclosure-disks.component.ts [hunk 1]
```diff
@@ -292,11 +286,11 @@ export class EnclosureDisksComponent implements AfterContentInit, OnDestroy {
     this.themeUtils = new ThemeUtils();
     this.diskTemperatureService.listenForTemperatureUpdates();
 
-    core.register({ observerClass: this, eventName: 'DiskTemperatures' }).pipe(untilDestroyed(this)).subscribe((evt: DiskTemperaturesEvent) => {
+    this.diskTemperatureService.temperature$.pipe(untilDestroyed(this)).subscribe((data) => {
```
Comment (RehanY147) [inline src/app/pages/system/view-enclosure/components/enclosure-disks/enclosure-disks.component.ts:289]: We shouldn't make the subject public. Instead, let's make the subject private and make an observable of the subject public.

**Review by RehanY147 — CHANGES_REQUESTED** [review]
This logic is very complicated to understand. I think since we are in the area, we can do a refactor of the `DiskUpdateService` and make it a store. Also, make the logic in `DiskTemperaturesService` a bit more easier to understand.

**Comment by RehanY147** [comment]
@bvasilenko You can disregard my last review. I will create a new ticket for that.
Also, can you give me a more specific instructions on what to test for? I have a mock system, so I don't know how I am going to check for changing temperature updates etc.

**Comment by bvasilenko** [comment]
> You can disregard my last review. I will create a new ticket for that.

Ok!

> Also, can you give me a more specific instructions on what to test for? I have a mock system, so I don't know how I am going to check for changing temperature updates etc.

3. navigate to **System Settings > Enclosure**

image

4. switch to **Temperature** view
image

**Review by RehanY147 — APPROVED** [review]

**Review by RehanY147 — CHANGES_REQUESTED** [review]
Same request is being made twice every N seconds. Plase review.

**Comment by bvasilenko** [comment]
@RehanY147 now requested changes should be done

**Review by RehanY147 — APPROVED** [review]

**Pull request merged**